src/main/java/jenkins/plugins/git/AbstractGitSCMSource.java

```diff
@@ -819,12 +836,12 @@ public SCMRevision run(GitClient client, String remoteName) throws IOException,
                                       if (branches.isEmpty()) {
                                           listener.getLogger().printf("Could not find a branch containing commit %s%n",
                                                   hash);
-                                          return null;
+                                          return new GitRefSCMRevision(new GitRefSCMHead(hash, hash), hash);
```

Comment:
cf. #557

Reply:
Not actually. This just touches one case, namely where the hash is in the remote repo but has not been GC'd yet

There are still issues with the fetching of the ref as we need to make changes to GitSCMBuilder to ensure that the required ref is fetched in the trimmed refspec